### PR TITLE
Remove null check from CCS APIs

### PR DESCRIPTION
**Pull request by trwalke**
Fixes # .
Fix for https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2755

**Changes proposed in this request**
Removes null check from WithCCS apis

**Testing**
Adding unit test

**Performance impact**
None

### CODE REVIEW

**Review by jennyf19 — APPROVED** [review]
:shipit:

**Comment by jennyf19** [comment]
we don't have any tests which cover this? maybe need to add one that shows the header not getting added?

**Comment by trwalke** [comment]
> we don't have any tests which cover this? maybe need to add one that shows the header not getting added?

Yes, im in the process of adding exactly that. 
thanks

**Comment by trwalke** [comment]
@jennyf19 added test

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @trwalke

File: src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenByAuthorizationCodeParameterBuilder.cs [hunk 1]
```diff
@@ -114,13 +114,9 @@ public AcquireTokenByAuthorizationCodeParameterBuilder WithPkceCodeVerifier(stri
         /// <returns>The builder to chain the .With methods</returns>
         public AcquireTokenByAuthorizationCodeParameterBuilder WithCcsRoutingHint(string userObjectIdentifier, string tenantIdentifier)
         {
-            if (string.IsNullOrEmpty(userObjectIdentifier))
+            if (string.IsNullOrEmpty(userObjectIdentifier) || string.IsNullOrEmpty(tenantIdentifier))
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenByAuthorizationCodeParameterBuilder.cs:117]: I guess CCS work with ADFS and B2C?

File: tests/Microsoft.Identity.Test.Common/Core/Mocks/MockHttpMessageHandler.cs [hunk 2]
```diff
@@ -22,6 +22,7 @@ internal class MockHttpMessageHandler : HttpMessageHandler
         public IDictionary<string, string> ExpectedQueryParams { get; set; }
         public IDictionary<string, string> ExpectedPostData { get; set; }
         public IDictionary<string, string> ExpectedRequestHeaders { get; set; }
+        public IDictionary<string, string> UnexpectedRequestHeaders { get; set; }
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Common/Core/Mocks/MockHttpMessageHandler.cs:25]: You don't do anything with the value, just with the key. So this should be a list? 

Alternatively, you can just inspect the `ActualRequestMessage`  in your test and assert directly. This doesn't look like a helper that we'd use too much.

File: tests/Microsoft.Identity.Test.Unit/RequestsTests/OboRequestTests.cs [hunk 3]
```diff
@@ -134,5 +138,32 @@ public async Task AcquireTokenByOboAccessTokenInCacheTestAsync()
...
+                var cca = ConfidentialClientApplicationBuilder
+                                                         .Create(TestConstants.ClientId)
+                                                         .WithClientSecret(TestConstants.ClientSecret)
+                                                         .WithAuthority(TestConstants.AuthorityCommonTenant)
+                                                         .WithHttpManager(httpManager)
+                                                         .BuildConcrete();
+
+                UserAssertion userAssertion = new UserAssertion(TestConstants.DefaultAccessToken);
+                var result = await cca.AcquireTokenOnBehalfOf(TestConstants.s_scope, userAssertion)
+                                      .WithCcsRoutingHint("")
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Unit/RequestsTests/OboRequestTests.cs:160]: I think you should call these methods twice for the test to be more roboust, i.e. 

AcquireToken.WithCcs("").Execute();
AcquireToken.WithCcs("", "").Execute();

**Comment by bgavrilMS** [comment]
LGTM, but please change the Mock helper./

**Review by bgavrilMS — APPROVED** [review]
Approved, but please see comments.

File: src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenByAuthorizationCodeParameterBuilder.cs (hunk 1 above)
Comment (jennyf19) [reply src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenByAuthorizationCodeParameterBuilder.cs:117]: ADFS uses it, B2C will ignore it, but there is another issue for [B2C](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2748)

File: tests/Microsoft.Identity.Test.Unit/PublicApiTests/ConfidentialClientApplicationTests.cs [hunk 4]
```diff
@@ -761,12 +761,37 @@ public async Task GetAuthorizationRequestUrl_NoHint_NoCcsRoutingHint_TestAsync()
...
-            Assert.AreEqual(ccsHint, qp[Constants.CcsRoutingHintHeader]);
+
+            if (!string.IsNullOrEmpty(ccsHint))
+            {
+                Assert.IsTrue(qp.ContainsKey(Constants.CcsRoutingHintHeader));
+                Assert.AreEqual(ccsHint, qp[Constants.CcsRoutingHintHeader]);
+            }
+            else
+            {
+                Assert.IsTrue(!qp.ContainsKey(Constants.CcsRoutingHintHeader));
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Unit/PublicApiTests/ConfidentialClientApplicationTests.cs:793]: `Assert.IsFalse(qp.ContainsKey(Constants.CcsRoutingHintHeader));`?

**Review by pmaytak — APPROVED** [review]

File: tests/Microsoft.Identity.Test.Common/Core/Mocks/MockHttpMessageHandler.cs (hunk 2 above)
Comment (trwalke) [reply tests/Microsoft.Identity.Test.Common/Core/Mocks/MockHttpMessageHandler.cs:25]: Yes. makes sense. just a list

File: tests/Microsoft.Identity.Test.Unit/PublicApiTests/ConfidentialClientApplicationTests.cs (hunk 4 above)
Comment (trwalke) [reply tests/Microsoft.Identity.Test.Unit/PublicApiTests/ConfidentialClientApplicationTests.cs:793]: This is an assertion to verify that the CCS hint is not added if the WithCcs... api is given a null value

**Pull request merged**